fearless/Modules/Staking/StakingMain/StakingMainPresenter.swift

```diff
@@ -95,16 +96,49 @@ extension StakingMainPresenter: StakingMainPresenterProtocol {
     }
 
     func performMainAction() {
-        let bonded = stateMachine.viewState { (_: BondedState) in true } ?? false
+        guard let bonded = stateMachine.viewState(using: { (state: BondedState) in state }) else {
+            wireframe.showSetupAmount(from: view, amount: amount)
+            return
+        }
 
-        if bonded {
-            if let stashItem = stateMachine.viewState(using: { (state: BondedState) in state.stashItem }) {
-                interactor.fetchController(for: stashItem.controller)
-            } else {
-                logger?.warning("Unexpected state on main action")
+        guard let controllerAccount = controllerAccount else {
+            if let view = view {
+                let locale = view.localizationManager?.selectedLocale
+                let controllerAddress = bonded.stashItem.controller
+                wireframe.presentMissingController(from: view, address: controllerAddress, locale: locale)
             }
-        } else {
-            wireframe.showSetupAmount(from: view, amount: amount)
+
+            return
+        }
+
+        let optExistingBonding: ExistingBonding? = stateMachine.viewState { (state: BondedState) in
```

Comment:
Looks like we already have state in line 99 so we don't need to query state machine one more time

Reply:
🤦‍♂️ sure!